### Perkenalan

=================================================

Graded Challenge 3

Nama  : Dwianti Hasanah

Batch : BSD-01


Assignment Graded Challenge ini dibuat untuk mengevaluasi hasil pembelajaran dalam minggu sebelumnya.
Adapun materi pembelajarannya adalah sebagai berikut.
  - Web Scraping
  - API
  - Business Knowledge
  - Statistic Descriptive
  - Statistic Inferential

Adapun konsep yang harus kita evaluasi dalam graded challege ini adalah sebagi berikut:
  - pengambilan data dari internet dengan web scraping
  - melakukan data preparation sebelum menlanjutkan analysis
  - membuat problem statement dengan menggunakan framework SMART
  - melakukan perhitungan Statistic Descriptive
  - pengujian statistik dan merumuskan hipotesis
  - membuat isight ari hasil perhitungan
  - membuat kesimpulan dalam menjawab problem statement
  
##### Dengan Kata lain mampu menerapkan dan mengoperasikan web scraping, data preparation, business understanding dengan menggunakan framework SMART, dan melakukan perhitungan statistik deskriptip
==================================================

### Case

### A. Web Scraping

Saya sekarang kekurangan uang. Untuk menambah penghasilan selain berkerja kantoran, saya memiliki ide untuk berjualan. Namun saya tidak memiliki cukup modal untuk memproduksi barang, uang yang saya miliki hanya cukup untuk melakukan promosi. 

Setelah melakukan browsing diinternet dan bertanya ke beberapa teman, sayang menemukan solusinya yaitu dengan melakukan dropshiper di platform Tokopedia. Saat saya masih bingung tentang produk apa yang ingin saya jual, saya teringat tentang yang sedang viral diakhir akhir ini. yaitu seblak, tapi saya tidak tau apakah orang orang memiliki animo yang besar terhadap seblak.

Dengan kemampuan yang saya miliki dari mengikuti bootcamp Data Science Hacktiv8 saya bisa menganalisis informasi tentang penjualan seblak di Tokopedia.

Tetapi dikarenakan saya tidak memiliki data untuk mengalisis saya harus melakukan Webscraping terlebih dahulu untuk mengambil data dan melakukan analisis.

  1. Melakukan pengambilan data dari website e-commerce Tokopedia
  2. Ambil data dengan `nama produk`, `harga produk`, lalu `lokasi toko`,`penjual`, `total terjual` dan `rating produk`.
  3. Ada skema promosi dalam website Tokopedia di panel teratas (produk yang membayar iklan). Ambillah produk yang regular saja dengan elemen `<div class="css-974ipl">`
    
  Catatan: Untuk produk baru yang tidak memiliki total terjual dan rating produk hasil akhir akan berbeda jumlah data yang diperoleh (otomatis tidak ada informasi). Atasilah masalah ini dengan menghandle data agar pandas bisa mendeteksi sebagai missing value.

  (Elemen telah disediakan, lakukan cek kebenaran elemen terlebih dahulu sebelum digunakan)

  4. Ambilah minimal 10 halaman pencarian informasi produk
  5. Setiap murid harus memiliki minimal 50 data dari 10 halaman yang diakses (setiap murid memiliki data yang berbeda)
  6. simpan hasil dari scrape data ke pandas DataFrame, lalu lakukan pengolahan data. (dibebaskan penamaan column)

Langkah pertama sebelum memulai scraping, pertama-tama dengan melakukan Import Library. Disini Saya menggunakan Selenium, beautifulsoup dan jtidak lupa pandas

In [89]:
# import library
from scipy import stats
import numpy as np
import seaborn as sns
from selenium import webdriver
from bs4 import BeautifulSoup
import requests
import time
import pandas as pd

Setelah melakukan Import library kita bisa mulai pengambilan data ke platform tokopedia dan mencari kata kunci `seblak` dengan cara sebagai berikut

In [90]:
driver = webdriver.Chrome() # .Chorme dikarenakan menggunakan browser Chrome. bisa diubah dengan menyesuaikan browser yang digunakan
url="https://www.tokopedia.com/search?navsource=&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st="
driver.get(url)
html = driver.page_source

soup = BeautifulSoup(html, "html.parser")
print(soup.prettify()[:700])

<html data-rh="lang,translate" lang="id" translate="no">
 <head>
  <title>
   Jual seblak | Tokopedia
  </title>
  <meta charset="utf-8"/>
  <meta content="initial-scale=1, minimum-scale=1, maximum-scale=5, user-scalable=yes, width=device-width" name="viewport"/>
  <link crossorigin="" href="https://assets.tokopedia.net" rel="preconnect"/>
  <link href="https://assets.tokopedia.net" nonce="" rel="dns-prefetch"/>
  <link crossorigin="" href="https://gql.tokopedia.com" rel="preconnect"/>
  <link href="https://gql.tokopedia.com" nonce="" rel="dns-prefetch"/>
  <link crossorigin="" href="https://googleads.g.doubleclick.net" rel="preconnect"/>
  <link href="https://googleads.g.doubleclick.net" no


Pengambilan data untuk nama produk, harga, penjual, lokasi toko, total terjualnya produk, dan rating dari produk.

Data yang diambil hanya produk yang regular ( produk yang berbayang tidak diambil)

Data diambil dari 10 halaman pencarian dengan dibatasi 25 data perhalaman. karena jika kita mengambil seluruh data dari satu halaman data akan lama dan terlalu banyak. maka dari itu saya membatasi hanya 25 data dari satu halaman dengan total 10 halaman

Setelah pengambilan data, Saya lengsung menyimpan hasil data ke pandas DataFramework untuk diolah kembali

In [292]:
Data = []

driver = webdriver.Chrome()
for i in range(1,11): # range(1,11) untuk mengakses 10 page 
    Tokped="https://www.tokopedia.com/search?navsource=&page={}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st=%22".format(i)

    driver.get(Tokped) 
    time.sleep(3) 
    html = driver.page_source 
    soup = BeautifulSoup(html, "html.parser") 

    for it in soup.findAll('div', class_="css-uwyh54")[:25]: # [:25] pengambilan berapa banyak dat adalam satu halaman
        nama  = it.find('div', class_="css-974ipl").get_text()
        harga = it.find('div', class_="css-1ksb19c").get_text()
        for it2 in it.findAll('div', class_="css-1rn0irl"):
            penjual = it2.findAll('span', class_="css-1kdc32b")[1].get_text()
            lokasi  = it2.findAll('span', class_="css-1kdc32b")[0].get_text()

        trjual = it.findAll('span', class_="css-1duhs3e")
        if len(trjual)>0:
            terjual = it.find('span', class_="css-1duhs3e").get_text()
        else:
            terjual = None
        rtng = it.findAll('span', class_="css-t70v7i")
        try:
            rating = it.find('span', class_="css-t70v7i").get_text()
        except:
            rating = None
      

        # Menyimpan hasil scrape ke pandas DataFrame
        Data.append(
                (nama, harga, penjual, lokasi, terjual, rating)
                )
#membuat column dengan hasil data dari scraping
df=pd.DataFrame(Data, columns=['Nama Produk', 'Harga', 'Penjual', 'Lokasi', 'Terjual', 'Rating'])

df

Nama Produk     Harga  \
0    Seblak Instan Ceu Nthien Khas Bandung Rasana N...  Rp15.000   
1    seblak instan jeletot jajanan hits bandung 1 p...   Rp4.180   
2    Parade DiskonKylafood Seblak Original PlayRp14...  Rp14.350   
3    seblak basreng pedas original exstra daun jeru...  Rp25.000   
4    Termurah Kerupuk Mentah-Kerupuk Seblak-Kerupuk...   Rp6.000   
..                                                 ...       ...   
245  Tokopedia NYAM!SEBLAK INSTANT TERMURAHRp3.999J...   Rp3.999   
246  Tokopedia NYAM!Seblak Instan Pedas Home MadeRp...   Rp3.500   
247  Seblak instan yoi enak dan kenyangRp7.500Bekas...   Rp7.500   
248  Tokopedia NYAM!Seblak Instan Komplit Pedas Gur...   Rp5.999   
249  SEBLAK INSTAN MAMANG TERMURAHRp4.200Jakarta Ba...   Rp4.200   

                            Penjual             Lokasi         Terjual Rating  
0          Central Seblak Nusantara  Tangerang Selatan   Terjual 1 rb+    4.9  
1             tokomakanankalimantan      Jakarta Barat   Terjual 1 rb+    4.8  
2                          kylafood            Bandung    Terjual 750+    5.0  
3                  seblak bandung99       Kab. Bandung   Terjual 2 rb+    4.8  
4                  Djuragan Kerupuk           Surabaya     Terjual 50+    5.0  
..                              ...                ...             ...    ...  
245  Pusat Cuankie & Cemilan Frozen      Jakarta Timur  Terjual 10 rb+    4.8  
246                      the Dhecip  Tangerang Selatan   Terjual 1 rb+    4.8  
247                       SeblakYoi             Bekasi    Terjual 250+    4.9  
248                  Baso Aci Ayang         Kab. Garut   Terjual 2 rb+    4.9  
249                global cookware1      Jakarta Barat     Terjual 80+    5.0  

[250 rows x 6 columns]

Disini saya menyimpan data ke dalam format CSV agar data tidak repot jika sewaktu adanya error saat menjalankan data

In [300]:
#menyimpan data ke csv
df.to_csv('Web Scraping.csv',index = False)
df

Nama Produk     Harga  \
0    Seblak Instan Ceu Nthien Khas Bandung Rasana N...  Rp15.000   
1    seblak instan jeletot jajanan hits bandung 1 p...   Rp4.180   
2    Parade DiskonKylafood Seblak Original PlayRp14...  Rp14.350   
3    seblak basreng pedas original exstra daun jeru...  Rp25.000   
4    Termurah Kerupuk Mentah-Kerupuk Seblak-Kerupuk...   Rp6.000   
..                                                 ...       ...   
245  Tokopedia NYAM!SEBLAK INSTANT TERMURAHRp3.999J...   Rp3.999   
246  Tokopedia NYAM!Seblak Instan Pedas Home MadeRp...   Rp3.500   
247  Seblak instan yoi enak dan kenyangRp7.500Bekas...   Rp7.500   
248  Tokopedia NYAM!Seblak Instan Komplit Pedas Gur...   Rp5.999   
249  SEBLAK INSTAN MAMANG TERMURAHRp4.200Jakarta Ba...   Rp4.200   

                            Penjual             Lokasi         Terjual  Rating  
0          Central Seblak Nusantara  Tangerang Selatan   Terjual 1 rb+     4.9  
1             tokomakanankalimantan      Jakarta Barat   Terjual 1 rb+     4.8  
2                          kylafood            Bandung    Terjual 750+     5.0  
3                  seblak bandung99       Kab. Bandung   Terjual 2 rb+     4.8  
4                  Djuragan Kerupuk           Surabaya     Terjual 50+     5.0  
..                              ...                ...             ...     ...  
245  Pusat Cuankie & Cemilan Frozen      Jakarta Timur  Terjual 10 rb+     4.8  
246                      the Dhecip  Tangerang Selatan   Terjual 1 rb+     4.8  
247                       SeblakYoi             Bekasi    Terjual 250+     4.9  
248                  Baso Aci Ayang         Kab. Garut   Terjual 2 rb+     4.9  
249                global cookware1      Jakarta Barat     Terjual 80+     5.0  

[250 rows x 6 columns]

### B. Data Preparation

1. Melakukan explore data sederhana
  - menampilkan beberapa baris data (menuliskan insight)
  - menampilkan informasi dasar data (menuliskan insight dan langkah apa yang harus dilakukan selanjutnya dari insight yang saya dapatkan)
  - melakukan pengecheckan missing value (tuliiskan insight dari hasil yang didapat)
2. Melakukan cleaning Data
  - Handling missing values
  - melakukan penyesuaian isi data
  - Changes datatypes if the datatypes are not correct

Catatan: pada column Terjual, ubahlah jumlah produk terjual menjadi angka saja.

In [354]:
#Lakukan read CSV agar bisa mengolah data
df = pd.read_csv('Web Scraping.csv')
df

Nama Produk     Harga  \
0    Seblak Instan Ceu Nthien Khas Bandung Rasana N...  Rp15.000   
1    seblak instan jeletot jajanan hits bandung 1 p...   Rp4.180   
2    Parade DiskonKylafood Seblak Original PlayRp14...  Rp14.350   
3    seblak basreng pedas original exstra daun jeru...  Rp25.000   
4    Termurah Kerupuk Mentah-Kerupuk Seblak-Kerupuk...   Rp6.000   
..                                                 ...       ...   
245  Tokopedia NYAM!SEBLAK INSTANT TERMURAHRp3.999J...   Rp3.999   
246  Tokopedia NYAM!Seblak Instan Pedas Home MadeRp...   Rp3.500   
247  Seblak instan yoi enak dan kenyangRp7.500Bekas...   Rp7.500   
248  Tokopedia NYAM!Seblak Instan Komplit Pedas Gur...   Rp5.999   
249  SEBLAK INSTAN MAMANG TERMURAHRp4.200Jakarta Ba...   Rp4.200   

                            Penjual             Lokasi         Terjual  Rating  
0          Central Seblak Nusantara  Tangerang Selatan   Terjual 1 rb+     4.9  
1             tokomakanankalimantan      Jakarta Barat   Terjual 1 rb+     4.8  
2                          kylafood            Bandung    Terjual 750+     5.0  
3                  seblak bandung99       Kab. Bandung   Terjual 2 rb+     4.8  
4                  Djuragan Kerupuk           Surabaya     Terjual 50+     5.0  
..                              ...                ...             ...     ...  
245  Pusat Cuankie & Cemilan Frozen      Jakarta Timur  Terjual 10 rb+     4.8  
246                      the Dhecip  Tangerang Selatan   Terjual 1 rb+     4.8  
247                       SeblakYoi             Bekasi    Terjual 250+     4.9  
248                  Baso Aci Ayang         Kab. Garut   Terjual 2 rb+     4.9  
249                global cookware1      Jakarta Barat     Terjual 80+     5.0  

[250 rows x 6 columns]

In [355]:
df.info()
#untuk mengetahui informasi dasar data

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250 entries, 0 to 249
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Nama Produk  250 non-null    object 
 1   Harga        250 non-null    object 
 2   Penjual      250 non-null    object 
 3   Lokasi       250 non-null    object 
 4   Terjual      239 non-null    object 
 5   Rating       238 non-null    float64
dtypes: float64(1), object(5)
memory usage: 11.8+ KB


Dari menampilkan informasi diketahui jika datatype tidak sesuai untuk column Harga, dan Terjual. Lalu adanya missing Value pada column Terjual dan Rating.

Setelah mengetahui adanya type data yang tidak sesuai, missing value langkah yang kita harus lakukan adalah melakukan filling missing value dan melakukan perubahan dalam dtype.

Tetapi dalam column Harga dan terjual adanya string atau karakter tertentu yang menyembabkan tidak bisanya mengubah tipe data. 

Maka dari itu hal pertama yang haris dilakukan adalah mengubah isi data(mereplace string/karakter yang tidak diperlukan)

In [356]:
df['Rating'].fillna(0, inplace=True) 
# menghandle Missing value dengan mengisi menjadi angka 0

In [357]:
df['Terjual'].fillna(0, inplace=True)
# menghandle Missing value dengan mengisi menjadi angka 0

In [358]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250 entries, 0 to 249
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Nama Produk  250 non-null    object 
 1   Harga        250 non-null    object 
 2   Penjual      250 non-null    object 
 3   Lokasi       250 non-null    object 
 4   Terjual      250 non-null    object 
 5   Rating       250 non-null    float64
dtypes: float64(1), object(5)
memory usage: 11.8+ KB


Setelah filling missing value , maka tidak ada lagi missing value pada data

In [361]:
#fuction untuk mereplace +
def converter(val):
    val = str(val)
    val = val.replace('+', '')
    return val

df['Terjual'] = df['Terjual'].apply(converter)

In [362]:
#melakukan replace agar bisa mengubah tipe data
df['Terjual'] = df['Terjual'].str.replace('Terjual ', '', regex=True)
df['Terjual'] = df['Terjual'].str.replace('rb', '000', regex=True)
df['Terjual'] = df['Terjual'].str.replace(' ', '', regex=True)

Setelah mereplace string/karakter, langkah selanjutnya adalah mengubah tipe data pada column Terjual menjadi data yang sesuai, integer.

In [363]:
df['Terjual'] = df['Terjual'].astype(int)
df.dtypes

Nama Produk     object
Harga           object
Penjual         object
Lokasi          object
Terjual          int32
Rating         float64
dtype: object

In [365]:
#melakukan replace agar bisa mengubah tipe data
f['Harga'] = df['Harga'].str.replace('Rp', '',regex=True)

In [367]:
#fuction untuk mereplace .
def converter(value):
    value = str(value)
    value = value.replace('.', '')
    return value

df['Harga'] = df['Harga'].apply(converter)

Setelah mereplace string/karakter, langkah selanjutnya adalah mengubah tipe data pada column Harga menjadi data yang sesuai, integer.

In [370]:
df['Harga'] = df['Harga'].astype(int)
df.dtypes

Nama Produk     object
Harga            int32
Penjual         object
Lokasi          object
Terjual          int32
Rating         float64
dtype: object

### C. Business Understanding

Disini saya akan membuat problem statement dengan menggunakan SMART framework dalam satu kalimat

Permasalahan:
  - ingin menambah penghasilan tambahan
  - menjadi dropshiper

Strategi
  - melakukan analisa harga produk, peminat


SMART FrameWork

S = menambah penghasilan dengan menjadi dropshiper makanan seblak

M = menambah penghasilan tambahan dari 20% ke 50%

A = berdasarkan proses perencanaan yang baik dengan melakukan analisa harga dan minat terlebih dahulu, untuk mencapai target menambah penghasilan menjadi 50% bisa tercapai

R = dari hasil proses yang dilalui, saya bisa mengetahui untuk melanjutkan dalam meningkatkan penghasilan dengan menjual seblak atau tidak 

T = target pencapaian untuk menambah penghasilan menjadi 50% dengan menjual seblak dalam kurun waktu 3 bulan dengan ketentuan waktu dalam setiap proses

### D. Analysis

Untuk melakukan analisis agar tujuan tercapai, saya bisa melakukan perhitungan median dan mean, menghitung pendapatan, perbandingan harga bahan baku dan pengujian korelasi data 

Langkah pertama saya harus menghitung rata rata dan median dari data harga, bannyaknya produk terjual dan rating yang 

Lalu, langkah selanjutnya adalah melakukan pemikiran potensi pendapatan dalam menjual seblak perbulan menggunakan confidence interval

Setelah mengetahui potensi pendapatan, lakukan perbandingan harga bahan baku untuk diluar wilayah jabodetabek dan diluar jabodetabek karena beda lokasi, maka kemungkinan biayanya juag berbeda. disini saya menggunakan uji hipotesis

Dan yang terakhir melihat korelasi apakah peminat yang menyukai seblak dengan harga murah lebih banyak

#### Menghitung Mean, Median, standar deviasi, skewness, kurtosis

In [371]:
#kolom Harga
print('Mean :',df['Harga'].mean())
print('Median :',df['Harga'].median())
#standar deviasi
std = df['Harga'].std()
print('Standard Deviasi:',std)
#skewness
print(f"Skewness: {df['Harga'].skew()}")
#Kurtosis
print(f"Kurtosis: {df['Harga'].kurtosis()}")

Mean : 14939.688
Median : 12000.0
Standard Deviasi: 13275.009755124498
Skewness: 2.3923725242646285
Kurtosis: 7.104994886896586


Diketahui dari hasil melakukan perhitungan rata rata, harga berkisar 14rb dan median berkisar 12rb.

Dalam Perhitungan standar deviasi, skewness dan kurtosis pada kolom `harga` diketahui bahwa data memiliki banyak outlier 

In [372]:
#kolom Terjual
print('Mean :',df['Terjual'].mean())
print('Median :',df['Terjual'].median())
#standar deviasi
std = df['Terjual'].std()
print('Standard Deviasi:',std)
#skewness
print(f"Skewness: {df['Terjual'].skew()}")
#Kurtosis
print(f"Kurtosis: {df['Terjual'].kurtosis()}")

Mean : 1171.472
Median : 250.0
Standard Deviasi: 2611.4567203046076
Skewness: 2.9558156600121697
Kurtosis: 7.242267149864215


Diketahui dari hasil melakukan perhitungan rata rata, banyaknya Terjualnya suatu produk berkisar 1100 dan median berkisar 250.

Dalam Perhitungan standar deviasi, skewness dan kurtosis pada kolom `Terjual` diketahui bahwa data memiliki banyak outlier 

In [373]:
#kolom Rating
print('Mean :',df['Rating'].mean())
print('Median :',df['Rating'].median())
#standar deviasi
std = df['Rating'].std()
print('Standard Deviasi:',std)
#skewness
print(f"Skewness: {df['Rating'].skew()}")
#Kurtosis
print(f"Kurtosis: {df['Rating'].kurtosis()}")

Mean : 4.6316
Median : 4.9
Standard Deviasi: 1.0500341551147898
Skewness: -4.15806925753791
Kurtosis: 15.66185834247862


Diketahui dari hasil melakukan perhitungan rata rata, rating dari suatu produk berkisar 4,6 dari 5 dan median berkisar 4.9.

Dalam Perhitungan standar deviasi, skewness dan kurtosis pada kolom `Rating` diketahui bahwa data normal tetapi banyak outlier juga

#### Perhitungan minimum dam maximum by convidence interval

In [384]:
#melakukan perhitungan Min dan Max
std = df['Terjual'].std()
N = len(df)
low, up = stats.norm.interval(0.95,loc=df['Terjual'].mean(),scale=std/np.sqrt(N))
print('Lower Limit:',low)
print('Upper Limit:',up)


Lower Limit: 847.7584195353681
Upper Limit: 1495.1855804646318


In [387]:
harga=df['Harga'].mean()
Max_Keuntungan = harga*up
print('Max Keuntungan: ', Max_Keuntungan)

Min_Keuntungan = harga*low
print('Min Keuntungan: ', Min_Keuntungan)

Max Keuntungan:  22337606.074240495
Min Keuntungan:  12665246.287231505


Potensi keunntungan dalam penjualan maximum bisa mencapai 22juta rupiah dan minimum mencapai 12juta rupiah dalam sebualn menjual seblak

#### Hipothesis test

Sebelum melakukan hipotesis, saya akan melakukan perbandingan harga barang antara daerah Jabodetabek dan di luar Jabodetabek. Pertama-tama lakukan pengecekan nama lokasi yang berada di wilayah jabodetabek dalam column Lokasi dengan menampilkan unique colomn tersebut

In [409]:
df['Lokasi'].unique()

array(['Tangerang Selatan', 'Jakarta Barat', 'Bandung', 'Kab. Bandung',
       'Surabaya', 'Jakarta Pusat', 'Jakarta Timur', 'Magelang',
       'Kab. Garut', 'Jakarta Selatan', 'Tangerang', 'Kab. Tangerang',
       'Bekasi', 'Kab. Bogor', 'Kab. Bekasi', 'Jakarta Utara', 'Cimahi',
       'Depok', 'Makassar', 'Kab. Majalengka', 'Kab.Ciamis', 'Batam',
       'Kab. Tasikmalaya', 'Denpasar', 'Malang', 'Tasikmalaya',
       'Kab. Mojokerto'], dtype=object)

Lalu lakukan list untuk wilayah jabodetabek dan non jabodetabek

In [407]:
jabodetabek_list = ['Tangerang Selatan', 'Jakarta Barat',
       'Jakarta Pusat', 'Jakarta Timur',  'Jakarta Selatan', 'Tangerang', 'Kab. Tangerang',
       'Bekasi', 'Kab. Bogor', 'Kab. Bekasi', 'Jakarta Utara',
       'Depok']
       
jabodetabek = df[df['Lokasi'].isin(jabodetabek_list)]
nonJabodetabek = df[~df['Lokasi'].isin(jabodetabek_list)]
nonJabodetabek

Nama Produk  Harga  \
2    Parade DiskonKylafood Seblak Original PlayRp14...  14350   
3    seblak basreng pedas original exstra daun jeru...  25000   
4    Termurah Kerupuk Mentah-Kerupuk Seblak-Kerupuk...   6000   
8    KERUPUK MENTAH SERBA 5000 BAHAN SEBLAKRp6.000K...   6000   
25   Kerupuk Putih Ori 5 Kg (BALL) Kerupuk Seblak /...  75000   
..                                                 ...    ...   
232  SEBLAK INSTAN ORI/SEBLAK INSTAN KERUPUK/SEBLAK...   3899   
233  GRATIS ONGKIR SEBLAK INSTAN/SEBLAK BANDUNG/SEB...   3999   
234  Seblak umi 400gr super pedasRp31.000Cashback 2...  31000   
243  GURILEM MINI 1KG Siomay Kering Toping Baso Aci...  29999   
248  Tokopedia NYAM!Seblak Instan Komplit Pedas Gur...   5999   

                          Penjual          Lokasi  Terjual  Rating  
2                        kylafood         Bandung      750     5.0  
3                seblak bandung99    Kab. Bandung     2000     4.8  
4                Djuragan Kerupuk        Surabaya       50     5.0  
8                        GK_Snack    Kab. Bandung      500     5.0  
25   Distributor Topping Baso Aci         Bandung        0     0.0  
..                            ...             ...      ...     ...  
232                     dakar_ofc  Kab. Mojokerto       70     5.0  
233                    MarkCi .id      Kab. Garut      100     4.3  
234             seblakumiofficial         Bandung       26     5.0  
243                   Pedas Gilaa        Surabaya      500     4.9  
248                Baso Aci Ayang      Kab. Garut     2000     4.9  

[69 rows x 6 columns]

Setelah itu lakukan Hitopetsis perbedaan harga di daerah jabodetabek dan diluar jabodetabek

Disini saya melakukan Hipotesis menggunakan tipe two sample independent test dikarenakan adanya dua wilayah yang ingin diketahui perkiraan dari harga bahan baku

In [408]:
t_stat, p_val = stats.ttest_ind(jabodetabek['Harga'],nonJabodetabek['Harga'])
print('T-Statistic:',t_stat)
print('P-value:',p_val) 

T-Statistic: -3.0445209628745937
P-value: 0.002581636029189141


#### Correlation Analysis test

In [397]:
#korelasi spearman
corr_rho, pval_s = stats.spearmanr(df['Harga'], df['Rating'])
print(f"Spearman correlation: {corr_rho:.2f}, p-value: {pval_s}")

Spearman correlation: 0.12, p-value: 0.06775764549908879


### E. Conclusion

Kesimpulan dari menganalisis data bisa diketahui:
  - Jika rata rata harga adalah sekitar Rp. 14.000,- dengan sekitar 1.000 produk yang terjual dengan rating 4.6
  - Median harga sekitar Rp. 12.000,- dengan terjualnya produk sekitar 250 dengan rating 4.9
  - Dalam test normalitas skewness dan kurtosis diketahui jika data tidak normal normal dan memiliki banyak outlier
  - Dalam menganalisa potensi keuntungan minimum dam maximum dalam penjualan seblak selama satu bulan cukup tinggi dengan kata lain bisa menghasilkan pendapatan tambahan dalam penjualan ini
  - Dapat disimpulkan jika harga bahan baku di wilayah Jabodetabek sangat berbeda dengan di luar jabodetabek menggunakan two sample independent test
  - Dalam data yang saya olah, saya menggunakan library spicy untuk menganalisis korelasi dan p-value menggunakan teknik spearman yang mana hasil menunjukkan jika korelasi antara cenderungnya orang suka dengan harga yang murah.
